src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -0,0 +1,325 @@
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
+                columnNumber);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Arrays.asList(
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK/METHOD_DEF[@text='Label']"
+                    + "/SLIST/LITERAL_SWITCH/CASE_GROUP",
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK/METHOD_DEF[@text='Label']"
+                    + "/SLIST/LITERAL_SWITCH/CASE_GROUP/LITERAL_DEFAULT");
+        assertEquals("Generated queries do not match expected ones", expected, actual);
+    }
+
+    @Test
+    public void testVariableStringLiteral() {
+        final int lineNumber = 46;
+        final int columnNumber = 25;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
+                columnNumber);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Arrays.asList(
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK"
+                    + "/METHOD_DEF[@text='callSomeMethod']/SLIST/VARIABLE_DEF[@text='another']"
+                    + "/ASSIGN/EXPR",
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK"
+                    + "/METHOD_DEF[@text='callSomeMethod']/SLIST/VARIABLE_DEF[@text='another']"
+                    + "/ASSIGN/EXPR/STRING_LITERAL");
+        assertEquals("Generated queries do not match expected ones", expected, actual);
+    }
+
+    @Test
+    public void testComma() {
+        final int lineNumber = 65;
+        final int columnNumber = 35;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
+                columnNumber);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Collections.singletonList(
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK/METHOD_DEF[@text='foo']"
+                    + "/SLIST/LITERAL_FOR/FOR_ITERATOR/ELIST/COMMA");
+        assertEquals("Generated queries do not match expected ones", expected, actual);
+    }
+
+    @Test
+    public void testLiteralVoid() {
+        final int lineNumber = 64;
+        final int columnNumber = 11;
+        final XpathQueryGenerator queryGenerator = new XpathQueryGenerator(rootAst, lineNumber,
+                columnNumber);
+        final List<String> actual = queryGenerator.generate();
+        final List<String> expected = Arrays.asList(
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK"
+                    + "/METHOD_DEF[@text='foo']/TYPE",
+            "/CLASS_DEF[@text='InputXpathQueryGenerator']/OBJBLOCK"
+                    + "/METHOD_DEF[@text='foo']/TYPE/LITERAL_VOID");
+        assertEquals("Generated queries do not match expected ones", expected, actual);
+    }
+
+    @Test
+    public void testNotUniqueDetailAst() {
```

Comment:
This is a synthetic example. If you can not prove that we need ```getFirstSibling``` with a real compilable code in input file, then you should remove the method.